app/components/GuideReviewForm/index.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import Select from 'components/Select';
+
+import PageSection from 'components/PageSection';
+import BorderBottomDiv from 'components/shared/BorderBottomDiv';
+import Stars from 'components/Stars';
+
+class GuideReviewForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleButtonClick = this.handleButtonClick.bind(this);
+    this.state = {
+      isGuided: props.isGuided,
+    };
+  }
+
+  componentWillMount() {
+    if(this.props.isGuided) {
+      this.props.getGuideList();  // lifting up state: API call is handled by WriteReviewPage container
+    }
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    // Update if...
+    const doUpdate = ((nextState.isGuided !== this.state.isGuided) //...switching between yes/no options (affects disabled status for Select component)
+      || (nextProps.guideOptions.length !== this.props.guideOptions.length) //...or guide list is updated (affects Select component's options)
+      || (nextProps.selectedGuideId !== this.props.selectedGuideId) //...or a guide is selected (affects Select component and disabled status for Stars component)
+    );
+
+    return doUpdate;
+  }
+
+  componentWillUpdate(nextProps) {
+    // Do another API call if, somehow, the options array's length changes
+    if(this.props.guideOptions.length && nextProps.guideOptions.length !== this.props.guideOptions.length) {
+      console.log('componentWillUpdate newAPIcall:', nextProps.guideOptions);
+      this.props.getGuideList();  // lifting up state: API call is handled by WriteReviewPage container
+    }
+  }
+
+  handleButtonClick(e) {
+    // There are edge cases where clicking on the edge of a button can give undefined value
+    const isGuided = e.target.firstChild.value ? e.target.firstChild.value === "guided" : this.state.isGuided;
```

Comment:
This is fine since there's only 2 options, but in general we should try to avoid stringy-typed code

Reply:
I agree, but when I put `{true}` and `{false}` in the value fields, they became strings anyway. So I made them "guide" and "notGuided" to at least be explicit.